app/routes/profile.js

```diff
@@ -1,3 +1,16 @@
 import Route from '@ember/routing/route';
+import ENV from 'website-my/config/environment';
 
-export default class ProfileRoute extends Route {}
+export default class ProfileRoute extends Route {
+  async model() {
+    try {
+      const response = await fetch(`${ENV.BASE_API_URL}/users/self`, {
+        credentials: 'include',
+      });
+      const userData = await response.json();
+      return userData;
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
```

Comment:
What if the user doesn't have the cookie? API will respond with 401 in that case
That case doesn't seem to be handled here